src/main/java/com/investment/manager/controller/InvestmentController.java

```diff
@@ -39,9 +39,28 @@ public InvestmentDTO getById(@PathVariable("id") Long id) throws NotFoundExcepti
 	}
 
 	@GetMapping(produces = MediaType.APPLICATION_JSON_VALUE)
-	public List<InvestmentDTO> getAll() {
+	public Page<InvestmentDTO> getAll(@RequestParam(value = "page", required = false, defaultValue = "0") int page,
+			@RequestParam(value = "size", required = false, defaultValue = "10") int size) {
+
+		return investmentService.getAll(page, size);
+
+	}
+
+	@GetMapping(value = "/broker/{broker}", produces = MediaType.APPLICATION_JSON_VALUE)
+	public Page<InvestmentDTO> getByBroker(@RequestParam("broker") UserDTO broker,
+			@RequestParam(value = "page", required = false) int page,
+			@RequestParam(value = "size", required = false, defaultValue = "10") int size) throws Exception {
+
+		return investmentService.searchByBroker(broker, page, size);
+
+	}
+
+	@GetMapping(value = "/client/{id}", produces = MediaType.APPLICATION_JSON_VALUE)
+	public Page<InvestmentDTO> getByClient(@RequestParam("id") UserDTO client,
```

Comment:
Aqui meio que parece estar esperando só o ID mesmo, mas utilize @PathVariable("id") Long id; (se o tipo for Long)